### PR TITLE
fix(data): return None for Energon test split instead of aliasing validation

### PR DESCRIPTION
**Pull request by khazic**
## What

`EnergonProvider.build_datasets` returned the validation dataloader for **both** the validation and test slots:

```python
return (
    iter(dataset.train_dataloader()),
    iter(dataset.val_dataloader()),
    iter(dataset.val_dataloader()),   # <- test slot reuses validation
)
```

But `EnergonMultiModalDataModule.test_dataloader()` intentionally returns `None` and logs `"Multimodal dataloader test dataset split does not exist"` — there is no distinct test split.

Because a non-None test dataloader was returned, downstream:

```python
do_test = test_dataloader is not None and cfg.validation.eval_iters > 0
```

evaluates to `True`, so end-of-training test evaluation runs on the validation set and **reports validation metrics as test metrics**, hiding whether the model generalizes to a genuinely held-out split.

## Fix

Return `iter(dataset.test_dataloader())` if present, else `None` — honoring the datamodule's own contract. With `None`, `do_test` stays `False` and test evaluation is correctly skipped (the consumer in `data/loaders.py` already handles a `None` test dataloader).

## Test

Updated `test_init_and_build_datasets` to assert the test iterator is `None` and `test_dataloader()` is consulted; updated the shared mock helper to mirror the real datamodule (test split returns `None`).

## Notes

Local unit-test execution isn't possible on this machine (megatron.core / CUDA deps); ruff check and format pass.

### CODE REVIEW

**Comment by yaoyu-33** [comment]
/ok to test https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/4659/commits/570fcb2b12e3a0fab3005c639c2b8a4b592eb22f
